mmaction/core/evaluation/accuracy.py

```diff
@@ -39,12 +39,17 @@ def confusion_matrix(y_pred, y_real, normalize=None):
 
     label_set = np.unique(np.concatenate((y_pred, y_real)))
     num_labels = len(label_set)
-    label_map = {label: i for i, label in enumerate(label_set)}
-    confusion_mat = np.zeros((num_labels, num_labels), dtype=np.int64)
-    for rlabel, plabel in zip(y_real, y_pred):
-        index_real = label_map[rlabel]
-        index_pred = label_map[plabel]
-        confusion_mat[index_real][index_pred] += 1
+    max_label = label_set[-1]
+    label_map = np.zeros(max_label + 1, dtype=np.int64)
```

Comment:
I think it's OK to use a dictionary instead of np array here, as the original one:
```
label_map = {label: i for i, label in enumerate(label_set)}
```
so that 4 lines -> 1 line
any performance concern here?

Reply:
```python
label_map = {label: i for i, label in enumerate(label_set)}
y_pred_mapped = [label_map[i] for i in y_pred]
y_real_mapped = [label_map[i] for i in y_real]
```
This is much slower even than the original one.

Reply:
numbers would be appreciated

Reply:
That's right, the for loop is much slower.

Reply:
The proposal by kenny VS The original one before this PR:
for `classes = 10, length = 100, times = 10000`, 1.2643 vs 1.1462
for `classes = 10, length = 100, times = 10000`, 0.6333 vs 0.5554
even worse